simd/unit_tests/include/SIMDTesting_Ops.hpp

```diff
@@ -80,7 +80,7 @@ class absolutes {
   template <typename T>
   auto on_host(T const& a) const {
     if constexpr (std::is_signed_v<typename T::value_type>) {
-#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
+#if defined(KOKKOS_ENABLE_DEPRECATED_CODE_4) && !defined(KOKKOS_COMPILER_NVCC)
```

Comment:
Do we need to guard only against nvcc? Will this be problematic for hipcc and sycl compiler as well?

Reply:
It seems to work fine for `HIP`, `SYCL`, and `Clang+Cuda`.

Reply:
What is the issue here?

Reply:
> t seems, however, that the compiler doesn't see that the correct overload referred to in the Kokkos::Experimental math function is annotated as host-only function. Therefore, we never call the Experimental function for nvcc.

Reply:
I do not understand the answer.  Also why do we guard here and just use `Kokkos::abs` in the next changeset?

Reply:
#6487 should fix this for good.